internal/signalfx-agent/pkg/monitors/prometheusexporter/prometheus.go

```diff
@@ -45,13 +45,29 @@ type Config struct {
 	// (the default).
 	MetricPath string `yaml:"metricPath" default:"/metrics"`
 
+	// Control the log level to use if a scrape failure occurs when scraping
+	// a target. Modifying this configuration is useful for less stable
+	// targets. Only the debug, info, warn, and error log levels are supported.
+	ScrapeFailureLogLevel string `yaml:"scrapeFailureLogLevel" default:"error"`
+
 	// Send all the metrics that come out of the Prometheus exporter without
 	// any filtering.  This option has no effect when using the prometheus
 	// exporter monitor directly since there is no built-in filtering, only
 	// when embedding it in other monitors.
 	SendAllMetrics bool `yaml:"sendAllMetrics"`
 }
 
+func (c *Config) Validate() error {
+	var logLevels = []string{"debug", "info", "warn", "error"}
```

Comment:
This can be done in a switch on the lowercase value, post my `ParseLevel()` request

Reply:
Refactored

Reply:
If we're now accepting all logrus level can the comment be updated?

Reply:
Done